x-pack/legacy/plugins/uptime/public/components/settings/certificate_form.tsx

```diff
@@ -0,0 +1,160 @@
+import { selectDynamicSettings } from '../../state/selectors';
+
+type NumStr = string | number;
+
+export type OnFieldChangeType = (field: string, value?: NumStr) => void;
+
+export interface SettingsFormProps {
+  onChange: OnFieldChangeType;
+  formFields: DynamicSettings | null;
+  fieldErrors: any;
+  isDisabled: boolean;
+}
+
+export const CertificateExpirationForm: React.FC<SettingsFormProps> = ({
+  onChange,
+  formFields,
+  fieldErrors,
+  isDisabled,
+}) => {
+  const dss = useSelector(selectDynamicSettings);
+
+  return (
+    <>
+      <EuiTitle size="s">
+        <h3>
+          <FormattedMessage
+            id="xpack.uptime.sourceConfiguration.certificationSectionTitle"
+            defaultMessage="Certificate Expiration"
+          />
+        </h3>
+      </EuiTitle>
+      <EuiSpacer size="m" />
+      <EuiDescribedFormGroup
+        title={
+          <h4>
+            <FormattedMessage
+              id="xpack.uptime.sourceConfiguration.stateThresholds"
+              defaultMessage="Expiration State Thresholds"
+            />
+          </h4>
+        }
+        description={
+          <FormattedMessage
+            id="xpack.uptime.sourceConfiguration.stateThresholdsDescription"
+            defaultMessage="Set certificate expiration warning/error thresholds"
+          />
+        }
+      >
+        <EuiFormRow
+          describedByIds={['errorState']}
+          error={fieldErrors?.certificatesThresholds?.errorState}
+          fullWidth
+          helpText={
+            <FormattedMessage
+              id="xpack.uptime.sourceConfiguration.errorStateDefaultValue"
+              defaultMessage="The default value is {defaultValue}"
+              values={{
+                defaultValue: (
+                  <EuiCode>{defaultDynamicSettings?.certificatesThresholds?.errorState}</EuiCode>
+                ),
+              }}
+            />
+          }
+          isInvalid={!!fieldErrors?.certificatesThresholds?.errorState}
+          label={
+            <FormattedMessage
+              id="xpack.uptime.sourceConfiguration.errorStateLabel"
+              defaultMessage="Error state"
+            />
+          }
+        >
+          <EuiFlexGroup>
+            <EuiFlexItem grow={2}>
+              <EuiFieldNumber
+                data-test-subj={`error-state-threshold-input-${dss.loading ? 'loading' : 'loaded'}`}
+                fullWidth
+                disabled={isDisabled}
+                isLoading={dss.loading}
+                value={formFields?.certificatesThresholds?.errorState || ''}
+                onChange={({ currentTarget: { value } }: any) =>
+                  onChange(
+                    'certificatesThresholds.errorState',
+                    value === '' ? undefined : Number(value)
+                  )
+                }
+              />
+            </EuiFlexItem>
+            <EuiFlexItem grow={1}>
+              <EuiSelect options={[{ value: 'day', text: 'Days' }]} />
```

Comment:
Do we want more options here besides `day`? If not we may not want to use a Select here.